### PR TITLE
monitoring: T4418: Add output plugin azure-data-explorer

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Add output telegraf Plugin Azure Data Explorer

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4418

## Component(s) name

monitoring, telegraf
## How to test
```
set service monitoring telegraf azure-data-explorer authentication client-id 'some-clientID'
set service monitoring telegraf azure-data-explorer authentication client-secret 'SecRtetsdf'
set service monitoring telegraf azure-data-explorer authentication tenant-id 'some-tanantID'
set service monitoring telegraf azure-data-explorer database 'foo_data'
set service monitoring telegraf azure-data-explorer group-metrics 'table-per-metric'
set service monitoring telegraf azure-data-explorer url 'http://localhost.loc'
```
telegraf config:
```
vyos@tstrtr2# cat /run/telegraf/vyos-telegraf.conf | grep Azure -A 10
### Azure Data Explorer ###
[[outputs.azure_data_explorer]]
  ## The URI property of the Azure Data Explorer resource on Azure
  endpoint_url = "http://localhost.loc"

  ## The Azure Data Explorer database that the metrics will be ingested into.
  ## The plugin will NOT generate this database automatically, it's expected that this database already exists before ingestion.
  database = "foo_data"
  metrics_grouping_type = "TablePerMetric"
### End Azure Data Explorer ###
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/service_monitoring_telegraf.xml.in [hunk 1]
```diff
@@ -42,6 +42,74 @@
...
+                </properties>
+                <children>
+                  <node name="authentication">
+                    <properties>
+                      <help>Authentication parameters</help>
+                    </properties>
+                    <children>
+                      <leafNode name="client-id">
+                        <properties>
+                          <help>Application client id</help>
```
Comment (c-po) [inline interface-definitions/service_monitoring_telegraf.xml.in:57]: Should we add a validator here?

File: interface-definitions/service_monitoring_telegraf.xml.in [hunk 2]
```diff
@@ -42,6 +42,74 @@
...
+                        </properties>
+                      </leafNode>
+                      <leafNode name="client-secret">
+                        <properties>
+                          <help>Application client secret</help>
```
Comment (c-po) [inline interface-definitions/service_monitoring_telegraf.xml.in:66]: Should we add a validator here?

File: interface-definitions/service_monitoring_telegraf.xml.in [hunk 3]
```diff
@@ -42,6 +42,74 @@
...
+                        </properties>
+                      </leafNode>
+                      <leafNode name="tenant-id">
+                        <properties>
+                          <help>Set tenant id</help>
```
Comment (c-po) [inline interface-definitions/service_monitoring_telegraf.xml.in:75]: Should we add a validator here?

File: interface-definitions/service_monitoring_telegraf.xml.in [hunk 4]
```diff
@@ -42,6 +42,74 @@
...
+                        </properties>
+                      </leafNode>
+                    </children>
+                  </node>
+                  <leafNode name="database">
+                    <properties>
+                      <help>Remote database name [REQUIRED]</help>
+                      <valueHelp>
+                        <format>txt</format>
+                        <description>Remote database name</description>
```
Comment (c-po) [inline interface-definitions/service_monitoring_telegraf.xml.in:89]: Should we add a validator here?

File: interface-definitions/service_monitoring_telegraf.xml.in [hunk 5]
```diff
@@ -42,6 +42,74 @@
...
+                      </constraint>
+                    </properties>
+                    <defaultValue>table-per-metric</defaultValue>
+                  </leafNode>
+                  <leafNode name="table">
+                    <properties>
+                      <help>Name of the single table [Only if set group-metrics single-table]</help>
+                      <valueHelp>
+                        <format>txt</format>
+                        <description>Remote database name</description>
```
Comment (c-po) [inline interface-definitions/service_monitoring_telegraf.xml.in]: Should we add a validator here?

**Review by c-po — CHANGES_REQUESTED** [review]

**Comment by sever-sever** [comment]
To be honest, I'm not familiar with Azure and required formats
I tried to find format `client-id, client secret, tenant-id` but it was unsuccessful
https://github.com/MicrosoftDocs/dataexplorer-docs/tree/main/data-explorer

It takes time to find at least some format data

**Comment by c-po** [comment]
> To be honest, I'm not familiar with Azure and required formats I tried to find format `client-id, client secret, tenant-id` but it was unsuccessful https://github.com/MicrosoftDocs/dataexplorer-docs/tree/main/data-explorer
> 
> It takes time to find at least some format data

I guess alphanumeric is always a good idea to be as open as possible bit deny gancy german umlauts, what you think?

**Comment by sever-sever** [comment]
> I guess alphanumeric is always a good idea to be as open as possible bit deny gancy german umlauts, what you think?

Done

**Pull request merged**